Framework/src/Aggregator.cxx

```diff
@@ -0,0 +1,120 @@
+      mAggregatorConfig.customParameters[_key] = _value.data();
+    }
+  }
+
+  ILOG(Info, Devel) << "Creation of a new aggregator " << aggregatorName << ENDM;
+  for (const auto& [_key, dataSource] : configuration.get_child("dataSource")) { // loop through data sources
+    (void)_key;
+
+    if (auto sourceType = dataSource.get<std::string>("type"); sourceType == "Aggregator" || sourceType == "Check") {
+      auto sourceName = dataSource.get<std::string>("name");
+      ILOG(Info, Devel) << "   Found a source : " << sourceName << ENDM;
+
+      if (dataSource.count("QOs") == 0 || dataSource.get<std::string>("QOs") == "all") {
+        ILOG(Info, Devel) << "      (no QOs specified or specified as `all`)" << ENDM;
+        mAggregatorConfig.allMOs = true;
+        // fixme: this is a dirty fix. Policies should be refactored, so this check won't be needed.
+        if (mAggregatorConfig.policyType != "OnEachSeparately") {
+          mAggregatorConfig.policyType = "_OnGlobalAny";
+        }
+      } else {
+        for (const auto& qoName : dataSource.get_child("QOs")) {
+          auto name = std::string(sourceName + "/" + qoName.second.get_value<std::string>());
+          ILOG(Info, Devel) << "      - " << name << ENDM;
+          mAggregatorConfig.moNames.push_back(name);
+        }
+      }
+    }
+  }
+}
+
+void Aggregator::init()
+{
+  try {
+    ILOG(Info, Devel) << "Insantiating the user code for aggregator " << mAggregatorConfig.name
+                      << " (" << mAggregatorConfig.moduleName << ", " << mAggregatorConfig.className << ")" << ENDM;
+    mAggregatorInterface =
+      root_class_factory::create<AggregatorInterface>(mAggregatorConfig.moduleName, mAggregatorConfig.className);
+    mAggregatorInterface->setCustomParameters(mAggregatorConfig.customParameters);
+    mAggregatorInterface->configure(mAggregatorConfig.name);
+  } catch (...) {
+    std::string diagnostic = boost::current_exception_diagnostic_information();
+    ILOG(Fatal, Ops) << "Unexpected exception, diagnostic information follows:\n"
+                     << diagnostic << ENDM;
+    throw;
+  }
+
+  // Print setting
+  ILOG(Info, Ops) << mAggregatorConfig.name << ": Module " << mAggregatorConfig.moduleName << AliceO2::InfoLogger::InfoLogger::endm;
+  ILOG(Info, Ops) << mAggregatorConfig.name << ": Class " << mAggregatorConfig.className << AliceO2::InfoLogger::InfoLogger::endm;
+  ILOG(Info, Ops) << mAggregatorConfig.name << ": Detector " << mAggregatorConfig.detectorName << AliceO2::InfoLogger::InfoLogger::endm;
+  ILOG(Info, Ops) << mAggregatorConfig.name << ": Policy " << mAggregatorConfig.policyType << AliceO2::InfoLogger::InfoLogger::endm;
+  ILOG(Info, Ops) << mAggregatorConfig.name << ": QualityObjects : " << AliceO2::InfoLogger::InfoLogger::endm;
+  for (const auto& moname : mAggregatorConfig.moNames) {
+    ILOG(Info, Ops) << mAggregatorConfig.name << "   - " << moname << AliceO2::InfoLogger::InfoLogger::endm;
+  }
```

Comment:
I would also add indication if "all" objects are expected, for the clarity. Just an empty list of QOs might be misleading in such case.

Reply:
solved by removing this option, see other comment